doc/guides/modules.rst

```diff
@@ -40,6 +40,102 @@ Zephyr depends on several categories of modules, including but not limited to:
 This page summarizes a list of policies and best practices which aim at
 better organizing the workflow in Zephyr modules.
 
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the zephyrproject-rtos Github organization.
+
+* The module repository code base shall include a *module.yml* file.
+
+* Module repository names should follow the convention of using lowercase
+  letters and dashes instead of underscores. This rule will apply to all
+  new module repositories, except for repositories that are directly
+  tracking external projects hosted in Github; such module repositories
+  may be named as their external project counterparts.
+
+  .. note::
+
+     Existing module repositories that do not conform to the above convention
+     do not need to be renamed to comply with the above convention.
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
```

Comment:
Doers it mean that the whole upstream project should be replicated, even files with no interest for Zephyr (utilities, middleware, ... in case of vendor HALs for instance)?

Reply:
Also, does it apply to existing module repositories ? If not, can it be mentioned ?

Reply:
No, but the folder structure should be intact, so Git should work OK. It's OK for certain files or directories to be skipped but not moved or re-shuffled.

Reply:
It does apply to existing repositories, but it is a recommendation (should) so I don't think we need a sentence here :)